app.py

```diff
@@ -1467,51 +1533,55 @@ def export_single_trip(trip_id):
                     "imei": trip["imei"]
                 }
             }
-            
+
             geojson = {
                 "type": "FeatureCollection",
                 "features": [feature]
             }
-            
+
             return Response(
                 json.dumps(geojson),
                 mimetype="application/geo+json",
-                headers={"Content-Disposition": f"attachment;filename=trip_{trip_id}.geojson"}
+                headers={
+                    "Content-Disposition": f"attachment;filename=trip_{trip_id}.geojson"}
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider extracting common header construction pattern</b></div>
<div id="fix">

Consider extracting the common header construction pattern into a helper function to avoid duplication of the `Content-Disposition` header setup for different file formats.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<blockquote>Check the AI-generated fix before applying</blockquote>
<div id="code">


```
 @@ -1544,4 +1544,11 @@
 +def _get_export_headers(trip_id, extension):
 +    return {
 +        "Content-Disposition": f"attachment;filename=trip_{trip_id}.{extension}"
 +    }
 +
                 headers={
 -                    "Content-Disposition": f"attachment;filename=trip_{trip_id}.geojson"}
 +                    _get_export_headers(trip_id, 'geojson')}
 @@ -1573,2 +1580,2 @@
                 headers={
 -                    "Content-Disposition": f"attachment;filename=trip_{trip_id}.gpx"}
 +                    _get_export_headers(trip_id, 'gpx')}
```

</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/realronaldrump/everystreet-new/pull/96#issuecomment-2543418637>#fab5ef</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged